File: SavingsAccount.java
public class SavingsAccount extends BankAccount {
    private double interestRate;
    public SavingsAccount() {
        interestRate = 0;
        balance = 0;
        person_name = "";
    }

    public SavingsAccount(String n, double b, double i) {
        balance = b;
        person_name = n;
        interestRate = i;
    }

    public void addInterest() {
        double interest = interestRate * balance;
        this.deposit(interest);
    }
}

File: SavingsAccount.cpp
//
// Created by Ted Hoeller on 11/27/2018.
//

#include "SavingsAccount.h"
#include <string>

using namespace std;

SavingsAccount::SavingsAccount(std::string n, double b, double i) : BankAccount(n, b) {
    this->interest = i;
}

void SavingsAccount::addInterest() {
    double added = interest * getBalance();
    deposit(added);
}